api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -386,6 +386,7 @@ public DbUser submitDataUseAgreement(
     // TODO: Teardown/reconcile duplicated state between the user profile and DUA.
     dbUser.setDataUseAgreementCompletionTime(timestamp);
     dbUser.setDataUseAgreementSignedVersion(dataUseAgreementSignedVersion);
+    dbUser.setLastModifiedTime(timestamp);
```

Comment:
This feels misplaced. If the issue is that last_modified_time isn't getting updated when the user is saved, I think we should really be changing that value up within the UserServiceImpl.updateUserWithRetries method, which is where all user-modifying changes funnel through. And then this method should probably call updateUserWithRetries instead of userDao.save.

Reply:
Can we use the updated `updateUserWithRetries` method to avoid setting this here?